src/shared/resources/profile/profileSlice.ts

```diff
@@ -7,14 +10,23 @@ import type {
   IProfileState,
 } from 'src/shared/resources/profile';
 import type { MkActionType } from 'src/shared/resources/utils/sagaSlice';
+import type { IAltinnWindow } from 'src/types';
 import type { IProfile } from 'src/types/shared';
 
-const initialState: IProfileState = {
+export interface IUpdateSelectedAppLanguage {
+  selected: string;
+}
+
+const altinnWindow = window as Window as IAltinnWindow;
+const getLanguageStorageKey = (userId: number | undefined) => `selectedAppLanguage${altinnWindow.app}${userId ?? ''}`;
+
+export const initialState: IProfileState = {
   profile: {
     profileSettingPreference: {
       language: 'nb',
     },
   } as IProfile,
+  selectedAppLanguage: '',
```

Comment:
Maybe the default language should be:
```suggestion
  selectedAppLanguage: getLanguageStorageKey(undefined),
```
This will then support remembering the selected language for stateless app (no user logg in). But is it neccassary to store the selected language for a stateless? Maybe it is better to store this in sessionStorage...

Reply:
First of all, thanks for your great contribution.  🚀 I agree with you that we should use "`getLanguageStorageKey(undefined)`" to persist the selected language on stateless apps. It's more of a nice-to-have feature than a need-to-have for stateless apps if you ask me.

Since the user is not logged in and we do not know if the next user is the same person, it might make sense to persist this within session storage instead of local storage to ensure other persons/users get the default language.

Within this PR I think it's good enough **not** persist the language within stateless apps. Because if we set the key `selectedAppLanguage${altinnWindow.app}` and later on move this to session storage, we need to write logic to clean up local storage by removing the key.  :)

Reply:
I agree. It makes sense not to save it to localStorage if no user is logged inn.